lib/intl.ts

```diff
@@ -133,56 +197,95 @@ if ('formatRangeToParts' in IntlDateTimeFormat.prototype) {
   properties.formatRangeToParts = descriptor(formatRangeToParts);
 }
 
-DateTimeFormat.prototype = Object.create(IntlDateTimeFormat.prototype, properties);
+DateTimeFormatImpl.prototype = Object.create(IntlDateTimeFormat.prototype, properties);
 
-function resolvedOptions(this: typeof DateTimeFormat) {
+function resolvedOptions(this: DateTimeFormatImpl): Return['resolvedOptions'] {
   return this[ORIGINAL].resolvedOptions();
 }
 
-function adjustFormatterTimeZone(formatter, timeZone) {
+function adjustFormatterTimeZone(
+  formatter: globalThis.Intl.DateTimeFormat,
+  timeZone?: string
+): globalThis.Intl.DateTimeFormat {
   if (!timeZone) return formatter;
   const options = formatter.resolvedOptions();
   if (options.timeZone === timeZone) return formatter;
-  return new IntlDateTimeFormat(options.locale, { ...options, timeZone });
+  // Existing Intl isn't typed to accept Temporal-specific options, but will not
+  // break at runtime if we pass them. Also, the lib types for resolved options
+  // are less restrictive than the types for options. For example, `weekday` is
+  // `'long' | 'short' | 'narrow'` in options but `string` in resolved options.
+  // TODO: investigate why, and file an issue against TS if it's a bug.
+  return new IntlDateTimeFormat(options.locale, { ...(options as globalThis.Intl.DateTimeFormatOptions), timeZone });
 }
 
-function format(this: typeof DateTimeFormat, datetime, ...rest) {
+// TODO: investigate why there's a rest parameter here. Does this function really need to accept extra params?
```

Comment:
Maybe so that in the future extra params can be passed to format without us having to specify them? Though it clearly isn't wired up into formatRange where it probably should... For another PR I think :)